openedx_tagging/core/tagging/models/base.py

```diff
@@ -342,6 +342,70 @@ def get_filtered_tags(
 
         return tag_set.order_by("value", "id")
 
+    def add_tag(
+        self,
+        tag_value: str,
+        parent_tag_id: int | None = None,
+        external_id: str | None = None
+
+    ) -> Tag:
+        """
+        Add new Tag to Taxonomy. If an existing Tag with the `tag_value` already
+        exists in the Taxonomy, an exception is raised, otherwise the newly
+        created Tag is returned
+        """
+        self.check_casted()
+
+        if self.allow_free_text:
+            raise ValueError(
+                "add_tag() doesn't work for free text taxonomies. They don't use Tag instances."
+            )
+
+        if self.system_defined:
+            raise ValueError(
+                "add_tag() doesn't work for system defined taxonomies. They cannot be modified."
+            )
+
+        if self.tag_set.filter(value__iexact=tag_value).exists():
+            raise ValueError(f"Tag with value '{tag_value}' already exists for taxonomy.")
+
+        parent = None
+        if parent_tag_id:
+            parent = Tag.objects.get(id=parent_tag_id)
+
+        tag = Tag.objects.create(
+            taxonomy=self, value=tag_value, parent=parent, external_id=external_id
+        )
+
+        return tag
+
+    def update_tag(self, tag_id: int, tag_value: str) -> Tag:
+        """
+        Update an existing Tag in Taxonomy and return it. Currently only
+        supports updating the Tag's value.
+        """
+        self.check_casted()
+
+        if self.allow_free_text:
+            raise ValueError(
+                "update_tag() doesn't work for free text taxonomies. They don't use Tag instances."
+            )
+
+        if self.system_defined:
+            raise ValueError(
+                "update_tag() doesn't work for system defined taxonomies. They cannot be modified."
+            )
+
+        # Update Tag instance with new value
+        tag = self.tag_set.get(id=tag_id)
```

Comment:
Similarly here, we should catch the possible `Tag.DoesNotExist`, and ensure the view responds with a 404.

Reply:
Addressed as well, thanks!